src/main/java/se/liu/ida/hefquin/queryplan/executable/IntermediateResultBlock.java

```diff
@@ -15,5 +16,9 @@
 	 * Returns an iterator over the intermediate result elements in this block.
 	 */
 	Iterator<SolutionMapping> iterator();
-	
+
+	/**
+	 * Returns an ArrayList over the intermediate result elements in this block.
+	 */
+	List<SolutionMapping> ArrayList();
```

Comment:
Two comments about this:

First, please use the standard naming convention for method names. That is, method names begin with a lower case letter. Hence, it should be 'arrayList' in this case or, actually, better would be 'getArrayList'.

Second, I would not want to add this method to this interface. We may have implementations of this interface in which the solution mappings are not just in a list but in a more sophisticated data structure (e.g., some form of an index -- see #3). For such implementations, it would be an unnecessary and inefficient overhead to create a list that can be returned here. Hence, wherever you wanted to use this list, use the iterator instead.